### ISSUE TITLE
Incorrect output for en-GB unit/narrow list formatter

### ISSUE THREAD

**Issue by linusg** [unicode-org/icu4x#5107]
I'm trying to replicate the following:

```js
new Intl.ListFormat("en-GB", { style: "narrow", type: "unit" }).format(["foo", "bar", "baz"]);
// "foo bar baz"
```

This works correctly when using `ListFormatter::try_new_unit_with_length` and `ListLength::Narrow` with the `en` locale, but for `en-GB` I get `"foo, bar, baz"`.

It appears the relevant file when using the compiled data provider is https://github.com/unicode-org/icu4x/blob/main/provider/data/list/data/unit_list_v1_marker.rs.data, which is missing definitions for `en-*` entirely but has a `("en", __AS)` mapping.

**Comment by linusg** [comment]
The relevant CLDR data is here:

https://github.com/unicode-org/cldr/blob/e899b786b8003d9f7b984bf710beeffd940a7682/common/main/en_GB.xml#L11069-L11074

https://github.com/unicode-org/cldr/blob/e899b786b8003d9f7b984bf710beeffd940a7682/common/main/en.xml#L9969-L9974

**Comment by linusg** [comment]
This affects more combinations than just unit/narrow, e.g. and/{wide,short,narrow} all result in `"foo, bar, baz"` instead of `"foo, bar, and baz"`, `"foo, bar, & baz"`, and `"foo bar baz"`.

**Comment by sffc** [comment]
Can you post the Rust code you believe is misbehaving?

I added tests in https://github.com/unicode-org/icu4x/pull/5108 and I'm getting output consistent with CLDR XML.

**Comment by linusg** [comment]
I'm using the C API wrapped in Zig so a few layers of indirection, the code is here: https://codeberg.org/linusg/icu4zig/src/commit/295756bde1bd3ca7c6bcc0b3010f82f0a534158b/src/ListFormatter.zig#L121-L148. I'll try to make a Rust reproducer from that.

**Comment by robertbastian** [comment]
Which ICU4X version are you using?

**Comment by linusg** [comment]
1.5.0 with this build configuration: https://codeberg.org/linusg/zig-build-icu4x/src/commit/141f169f07b49e4c877db38455e60e1068ae7ecd/build.zig#L46-L72

**Comment by sffc** [comment]
I'm fine reading the Zig code but it would be useful if you could post "here is the code I ran, this is what I expected, but this is what I actually got". It's hard to tell from the test case you linked: I can't tell which cases pass and which fail just by looking at the code.

**Comment by linusg** [comment]
Yes, I'll get you a reproducer in plain C and/or Rust - please disregard the issue until then, I understand it's not actionable at the moment since you didn't manage to reproduce it with the additional tests :)

**Comment by linusg** [comment]
I'm able to reproduce this using the C bindings with the following code.

- `git checkout icu@1.5.0` to make the below compile (a few APIs changed since then on `main`)
- `cd tutorials/c/ && touch listformat.c`
- Add the code below
- `make`

This outputs:

```
en: foo bar baz
en-GB: foo, bar, baz
```

```c
#include "ICU4XDataProvider.h"
#include "ICU4XLocale.h"
#include "ICU4XListFormatter.h"
#include <stdio.h>

int main() {
    ICU4XDataProvider* provider = ICU4XDataProvider_create_compiled();

    ICU4XList* list = ICU4XList_create_with_capacity(3);
    ICU4XList_push(list, "foo", 3);
    ICU4XList_push(list, "bar", 3);
    ICU4XList_push(list, "baz", 3);

    {
        char output[40];
        DiplomatWriteable write = diplomat_simple_writeable(output, sizeof(output));

        ICU4XLocale* locale = ICU4XLocale_create_from_string("en", 2).ok;
        ICU4XListFormatter* list_formatter = ICU4XListFormatter_create_unit_with_length(provider, locale, ICU4XListLength_Narrow).ok;
    
        ICU4XListFormatter_format(list_formatter, list, &write);
        printf("en: %s\n", output);

        ICU4XListFormatter_destroy(list_formatter);
        ICU4XLocale_destroy(locale);
    }

    {
        char output[40];
        DiplomatWriteable write = diplomat_simple_writeable(output, sizeof(output));

        ICU4XLocale* locale = ICU4XLocale_create_from_string("en-GB", 5).ok;
        ICU4XListFormatter* list_formatter = ICU4XListFormatter_create_unit_with_length(provider, locale, ICU4XListLength_Narrow).ok;
    
        ICU4XListFormatter_format(list_formatter, list, &write);
        printf("en-GB: %s\n", output);

        ICU4XListFormatter_destroy(list_formatter);
        ICU4XLocale_destroy(locale);
    }

    ICU4XList_destroy(list);
    ICU4XDataProvider_destroy(provider);

    return 0;
}
```

**Comment by robertbastian** [comment]
This outputs
```
gcc -L target/debug -I /Users/robertbastian/Code/icu4x/ffi/capi/bindings/c listformat.c -licu_capi -lm -o listformat
./listformat
en: foo bar baz
en-GB: foo bar baz
``` 
for me.

Did you set the `ICU4X_DATA_DIR` environment variable?

**Comment by linusg** [comment]
> Did you set the `ICU4X_DATA_DIR` environment variable?

That's unset. Some other random bits of information: GCC 13.3.0, NixOS 24.11. If you have any suggestions for further debugging please tell, I'm at a loss.

**Comment by linusg** [comment]
Actually, that made me find the issue! I'm using `rustc 1.80.0-nightly (804421dff 2024-06-07)`, with stable `rustc 1.78.0 (9b00956e5 2024-04-29)` the issue disappears.

**Comment by robertbastian** [comment]
I can reproduce this in a Rust unit test with `nightly-2024-06-22`

**Comment by robertbastian** [comment]
The failure is due to corrupted fallback data, i.e. https://github.com/unicode-org/icu4x/issues/5039. This has already been fixed (it fails at the `icu@1.5.0` tag but works on the `release/1.5` maintenance branch), and was released today. Run `cargo update`.

**Comment by linusg** [comment]
Thanks!
